src/constructUrl.js

```diff
@@ -106,12 +105,18 @@ var DEFAULT_OPTIONS = Object.freeze({
 });
 
 function constructUrlFromParams(src, params) {
-  var optionKeys = Object.keys(params);
-  var fullUrl = optionKeys.reduce(function(uri, key) {
-    return uri.replaceQueryParam(key, params[key]);
-  }, new Uri(src));
-
-  return fullUrl.toString();
+  const [url, query] = src.split("?");
+  const oldParams = query ? query.split("&").map(x => x.split("=")) : [];
+  const allParams = {
+    ...oldParams.reduce(
+      (obj, [key, val]) => Object.assign(obj, { [key]: val }),
+      {}
+    ),
+    ...params
+  };
+  return `${url}?${Object.entries(allParams)
+    .map(([a, b]) => `${a}=${encodeURIComponent(b)}`)
```

Comment:
This would double-encode any parameters that came in pre-encoded in `src`, right? Perhaps those parameters should be naively decoded during the map step on L109.

Reply:
Yes. Good point. I didn’t think of that.